File: Prob4.js
// A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.

//Find the largest palindrome made from the product of two 3-digit numbers.
// js convert number to string / var n = num.toString();



var palArray = [];
var numberComps = 0;


function isPalindrome(number) {
    var numString = number.toString();
    var numLength = numString.length;
    for (var i = 0; i<numLength; i++) {
       palArray.push(numString.charAt(i));
        }
 
    // determine how many comparisons to make start to end
    // inside the numString
    
    // if it's an even number, compare half its length. If it's odd,
    // compare half (length -1 )
    
           if (numLength % 2 === 0) {
                numberComps = (numLength / 2);
            }
            else
            {
                numberComps = ((numLength-1) / 2);
            }
 
  //isp variable initialized to true.
    
    var isp = true;
    
    //look at each front to middle and back to middle, comparing them.
    // if it finds one that doesn't match, it's not a palindrome
 
    for (var front = 0; front < numberComps; front++) {
        
        var back = ((numLength-1) - front);

         if (numString[front] != numString[back]) {
            isp = false;
        }             
}
  
    return isp;
 };



//***** code to multiply two 3-digit numbers


var checkPal = 0;
var highestPal = 0;

//checking only from 900 to 1000 because it's not running
// too memory-intensive

for (var x = 900; x < 1000; x++) {
    for (var y = 900; y < 1000; y++) {
        checkPal = (x*y);
            if (isPalindrome(checkPal)) {
                highestPal = checkPal;
            }
    }
}

console.log("highest palindrome:");
console.log(highestPal);

//it worked! answer : 906609

File: Prob4.rb
=begin


A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.

Find the largest palindrome made from the product of two 3-digit numbers.
=end

i = 1
t = 1
highest = 0
while i <1000 do
    t= 0
    while t < 1000 do
        #puts (i*t).to_s
        if (i*t).to_s == (i*t).to_s.reverse
            if (i*t) > highest
                highest = (i*t)
            end
        end
        t = t+1
    end
    i=i+1
end
puts highest
# answer 906609

# this is much simpler and more elegant in ruby. Of course in the JS version
# I wrote out all the code for determining whether a string was a palindrome
